services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/schema/EntitySchemaValidatorImplTest.java

```diff
@@ -280,15 +280,32 @@ public void testContainsAccessRequirementIdsWithNestedNoMatch() {
 	
 	@Test
 	public void testSetDerivedAnnotationsAndBindAccessRequirements() {
+		when(mockDerivedAnnotationDao.getDerivedAnnotations(any())).thenReturn(Optional.empty());
+
 		// call under test
 		boolean changed = manager.setDerivedAnnotationsAndBindAccessRequirements(objectDescriptor, annotations,
 				accessRequirmentIdsToBind);
 		assertTrue(changed);
+		verify(mockDerivedAnnotationDao).getDerivedAnnotations(entityId);
 		verify(mockAccessRequirementManager).setDynamicallyBoundAccessRequirementsForSubject(objectDescriptor,
 				accessRequirmentIdsToBind);
 		verify(mockDerivedAnnotationDao).saveDerivedAnnotations(entityId, annotations);
 		verify(mockDerivedAnnotationDao, never()).clearDerivedAnnotations(any());
 	}
+
+	@Test
+	public void testSetDerivedAnnotationsAndBindAccessRequirementsWithExistingEqualAnnotations() {
```

Comment:
We also need a test where the existing ones are different (make sure you verify that we do actually save the new annotations in this case)